app/Http/Controllers/UserProfileController.php

```diff
@@ -34,9 +34,17 @@ public function show($id = null)
             ->with('committee')
             ->where('user_id', $user->id)
             ->whereNotIn('id', $user->committees->pluck('pivot.id'))
-            ->get();
+            ->get()
+            ->where('committee.is_society', false);
```

Comment:
If I'm correct less queries are run if the society check is done within the with statement.

Reply:
See above.